core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -679,12 +680,25 @@ private[spark] object Utils extends Logging {
    * If no directories could be created, this will return an empty list.
    */
   private[spark] def getOrCreateLocalRootDirs(conf: SparkConf): Array[String] = {
+    if (localRootDirs == null) {
```

Comment:
The problem with the lazy val is that you don't get access to the `SparkConf` argument.